app/models/repository_status.rb

```diff
@@ -0,0 +1,93 @@
+# frozen_string_literal: true
+
+class RepositoryStatus < ApplicationRecord
+  belongs_to :repository
+
+  after_create :perform_checks
+
+  def name
+    @name ||= repository.name
+  end
+
+  def full_name
+    @full_name ||= repository.full_name
+  end
+
+  def perform_checks
+    self.missing_in_plumbing = !RepoStatusData.plumbing_modules.include?(name)
+
+    self.need_initial_modulesync = (!RepoStatusData.modulesync_repos.include?(name) &&
+                                                  !LEGACY_OR_BROKEN_NOBODY_KNOWS.include?(name))
+
+    self.need_initial_release = (!RepoStatusData.forge_releases.include?(name) &&
+                                                  !LEGACY_OR_BROKEN_NOBODY_KNOWS.include?(name))
+
+    self.without_reference_dot_md = !Github.get_file(full_name, 'REFERENCE.md')
+    self.added_but_never_synced = !Github.get_file(full_name, '.msync.yml')
+
+    self.missing_in_modulesync_repo = !RepoStatusData.modulesync_repos.include?(name)
+
+    msync_file = Github.get_file(full_name, '.msync.yml')
+    if msync_file
+      self.never_synced = false
+      msync_version = YAML.safe_load(msync_file)['modulesync_config_version']
+
+      self.needs_another_sync = (
+        Gem::Version.new(RepoStatusData.latest_modulesync_version) >
+          Gem::Version.new(msync_version)
+      )
+    else
+      self.never_synced = false
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Move `self.never_synced = false` out of the conditional.](https://app.codacy.com/manual/VoxPupuli/vox-pupuli-tasks/pullRequest?prid=5444834)